src/mqueue.rs

```diff
@@ -4,142 +4,126 @@
 
 use {Errno, Result};
 
-use libc::{c_int, c_long, c_char, size_t, mode_t};
+use libc::{self, c_char,c_long,mode_t,mqd_t,size_t};
 use std::ffi::CString;
 use sys::stat::Mode;
-use std::ptr;
-
-pub use self::consts::*;
-
-pub type MQd = c_int;
-
-#[cfg(target_os = "linux")]
-mod consts {
-    use libc::c_int;
-
-    bitflags!(
-        flags MQ_OFlag: c_int {
-            const O_RDONLY    = 0o00000000,
-            const O_WRONLY    = 0o00000001,
-            const O_RDWR      = 0o00000002,
-            const O_CREAT     = 0o00000100,
-            const O_EXCL      = 0o00000200,
-            const O_NONBLOCK  = 0o00004000,
-            const O_CLOEXEC   = 0o02000000,
-        }
-    );
-
-    bitflags!(
-        flags FdFlag: c_int {
-            const FD_CLOEXEC = 1
-        }
-    );
+use std::mem;
+
+bitflags!{
+    flags MQ_OFlag: libc::c_int {
+        const O_RDONLY    = 0o00000000,
+        const O_WRONLY    = 0o00000001,
+        const O_RDWR      = 0o00000002,
+        const O_CREAT     = 0o00000100,
+        const O_EXCL      = 0o00000200,
+        const O_NONBLOCK  = 0o00004000,
+        const O_CLOEXEC   = 0o02000000,
```

Comment:
Yes, it would be better. I will change that.

It would be even better to use `OFlag` from `fcntl.rs`. But `Oflag` has more flags than are allowed for the mqueue functions. I suppose, the other flags would just be ignored if passed into the functions. And just to be safe we could warn about their usage in mqueue functions. I have not made my mind up yet, and this PR is just for getting rid of the `ffi` module. There is much more that could be done to clean up this module.